pkg/tfbridge/detailed_diff.go

```diff
@@ -0,0 +1,430 @@
+package tfbridge
+
+import (
+	"context"
+	"errors"
+	"slices"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/info"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+)
+
+func isPresent(val resource.PropertyValue) bool {
+	return !val.IsNull() &&
+		!(val.IsArray() && val.ArrayValue() == nil) &&
+		!(val.IsObject() && val.ObjectValue() == nil)
+}
+
+func isForceNew(tfs shim.Schema, ps *SchemaInfo) bool {
+	if tfs != nil && tfs.ForceNew() {
+		return true
+	}
+	if ps != nil && ps.ForceNew != nil && *ps.ForceNew {
+		return true
+	}
+	return false
+}
+
+func isObject(tfs shim.Schema, ps *SchemaInfo) bool {
```

Comment:
I also object to this (pun intended). Can we hide away this gobbledygook? Or perhaps I'm too naive and schema lookups don't return us what we want here?

Idea 1: maps aren't objects. Don't put them on the same case branch. 

Idea 2: we've won if this file does not mention MaxItemsOne=1 at all. Is it possible to outsource this to some helpers in the shim layer?

This boils down to diligent case analysis. When I'm looking at (tfs, ps) pair, and an object type might be involved in describing the Pulumi schema of this thing, this object type may be involved in these ways, this may be one of:

```
- {foo: number}
- List<{foo: number}>
- Set<{foo: number}>
- Map<string, {foo: number}>
```

Note for list/set/map case these are simply collections, and should not need to care that much. The subsequent recursion of schema lookup will be finding the collection element type to deal with the element and that should be it.

So what's really needed is being able to tell if (tfs, ps) represents an object directly. I thought we have this. Digging.

Reply:
We discussed this offline and agreed we'd go with the current implementation as there are some issues preventing us from removing the MaxItemsOne logic from the module.

Specifically, when looking up the TF schema for a property path which points to a MaxItemsOne flattened collection, we currently return the collection and not the element - this is in contrast to the pulumi value, which will point to the element.